### ISSUE TITLE
Migrate to more recent .NET version

### ISSUE THREAD

**Issue by nikeee** [nikeee/TeamSpeak3QueryAPI#75]
What benefits could we have when migrating to a more recent .NET version (most likely .NET 6)?

**Comment by nikeee** [comment]
- We could enable `#nullable` to make the library more null-safe
- `System.ValueTuple` and maybe `Microsoft.CSharp` could be removed from the dependencies
- What about `System.Runtime.Serialization.Formatters`?
- DTOs could be converted to `record`s, auto-implementing equality

**Comment by MaximilianAst** [comment]
* All of the syntactic sugar :)

**Comment by ceeno2k** [comment]
- Usage of `Span<T>` and `Memory<T>` to improve performance (mainly in `FileTransferClient`, see https://github.com/nikeee/TeamSpeak3QueryAPI/pull/51#discussion_r421788093)

**Comment by nikeee** [comment]
That response parser that uses reflection could be re-written to use a source generator.

**Comment by nikeee** [comment]
With TS5 around the corner, it is questionable if some of the work is worth investing. I doubt that TS5 will use the same crappy protocol. At least I hope that they will migrate to something more sane, like JSON-RPC.

**Comment by Splamy** [comment]
TS5 is only a client rework, the server is the same.

**Comment by nikeee** [comment]
That's what I heard as well, but is that assured?

It would actually be pretty pointless, if they'd just copy some UI stuff from discord. So I am assuming that some server functionality will be added as well.

**Comment by Splamy** [comment]
The keep a lot vague, but I was in the TS5 beta and the client connects just fine to TS3 servers.
I guess this will be the targeted path of them. It's less work and they won't split their already shrinking userbase in half with this change.
They most likely will just add supplementary commands like they did in the past with [notifychannelpermhints](https://github.com/ReSpeak/tsdeclarations/blob/master/Messages.toml#L449) and notifyclientpermhints to improve the TS5 UI but keep it on the old protocol.

**Comment by nikeee** [comment]
Could we still target .NET Standard 1.3? I know it's old, but there are a lot of users using .NET FW.

I'll evaluate if we can target Standard 2.0 and use more recent .NET features.

**Comment by Splamy** [comment]
Yeah, should also work, you're basically just limiting your access to the 'nice' newer apis, but other than that probably no problems.
